applepaydomain.go

```diff
@@ -6,6 +12,12 @@ type ApplePayDomainParams struct {
 	DomainName *string `form:"domain_name"`
 }
 
+// ApplePayDomainListParams are the parameters allowed during ApplePayDomain listing.
```

Comment:
OOC will we add real comments from the API ref like we do in other libraries in the future?

Reply:
Yes, I want to do that as a single commit